tests/v2/validation/provisioning/rke2/templates_test.go

```diff
@@ -0,0 +1,164 @@
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+
+	provider := provisioning.CreateProvider(r.templateConfig.TemplateProvider)
+	r.cloudCredentials, err = provider.CloudCredFunc(client)
+	require.NoError(r.T(), err)
+}
+
+func (r *ClusterTemplateTestSuite) TestProvisionRKE2TemplateCluster() {
+	_, err := r.client.Catalog.ClusterRepos().Create(context.TODO(), r.templateConfig.Repo, metav1.CreateOptions{})
+	require.NoError(r.T(), err)
+
+	watchRepo, err := r.client.Catalog.ClusterRepos().Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + r.templateConfig.Repo.Name,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	require.NoError(r.T(), err)
+
+	err = wait.WatchWait(watchRepo, func(event watch.Event) (ready bool, err error) {
+		if event.Type == watch.Added {
+			return true, nil
+		} else if event.Type == watch.Error {
+			return false, fmt.Errorf("there was an error waiting for repo to become active")
+		}
+		return false, nil
+	})
+	require.NoError(r.T(), err)
+
+	r.client.Session.RegisterCleanupFunc(func() error {
+		err := r.client.Catalog.ClusterRepos().Delete(context.TODO(), r.templateConfig.Repo.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+
+		watchRepo, err := r.client.Catalog.ClusterRepos().Watch(context.TODO(), metav1.ListOptions{
+			FieldSelector:  "metadata.name=" + r.templateConfig.Repo.Name,
+			TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+		})
+		if err != nil {
+			return err
+		}
+
+		err = wait.WatchWait(watchRepo, func(event watch.Event) (ready bool, err error) {
+			if event.Type == watch.Error {
+				return false, fmt.Errorf("there was an error while removing repository")
+			} else if event.Type == watch.Deleted {
+				return true, nil
+			}
+			return false, nil
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	latestVersion, err := r.client.Catalog.GetLatestChartVersion(r.templateConfig.TemplateName, r.templateConfig.Repo.ObjectMeta.Name)
+	require.NoError(r.T(), err)
+
+	project, err := projects.GetProjectByName(r.client, localCluster, "System")
+	require.NoError(r.T(), err)
+
+	installOptions := &charts.InstallOptions{
+		Cluster: &clusters.ClusterMeta{
+			ID: localCluster,
+		},
+		Version:   latestVersion,
+		ProjectID: project.ID,
+	}
+
+	err = charts.InstallRKE2TemplateChart(r.client, installOptions, r.templateConfig.Repo.ObjectMeta.Name, r.templateConfig.TemplateName, r.cloudCredentials)
+	require.NoError(r.T(), err)
+
+	_, cluster, err := clusters.GetProvisioningClusterByName(r.client, r.templateConfig.TemplateName, fleetNamespace)
```

Comment:
Does `InstallRKE2TemplateChart` install chart + deploy cluster? Because if I understand it correctly `GetProvisioningClusterByName` returns the cluster deployed.

Reply:
Installing the chart deploys the cluster as you mentioned, however since it doesn't return the cluster object I need to get the cluster that was created, so I can run verify checks on it.

Reply:
Can you clarify if as part of this line, the cluster is deployed? - https://github.com/rancher/shepherd/pull/197/files#diff-a3ad61fa2ac31ef79c1ecfb8b3b02f298c7f254712a927cf849e9f6a7daf20f5R59